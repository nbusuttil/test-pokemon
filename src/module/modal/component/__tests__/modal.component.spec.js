import React from 'react';
import { shallow } from 'enzyme';

import Modal from '../modal.component';

describe('Modal component', () => {
  let props;

  beforeEach(() => {
    props = {
      loadPokemonInfo: jest.fn(),
      resetPokemonInfo: jest.fn(),
      closeModal: jest.fn(),
      pokemon: {
        id: 1,
        price: 4,
        quantity: 0,
        name: 'fearow',
        url: 'https://pokeapi.co/api/v2/pokemon/22/'
      }
    };
  });
  const getWrapper = () => shallow(<Modal {...props} />);

  it('render without crashing', () => {
    expect(getWrapper()).toMatchSnapshot();
  });
});
