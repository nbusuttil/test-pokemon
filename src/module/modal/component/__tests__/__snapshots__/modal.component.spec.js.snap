// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal component render without crashing 1`] = `
<div
  className="modal"
>
  <div
    className="modal__container"
  >
    <div
      className="modal__container-header"
    >
      <button
        onClick={[Function]}
      >
        X
      </button>
    </div>
    <div
      className="modal__container-content"
    >
      <h1>
        fearow
         
        <span>
          Pokedex: #
          1
        </span>
      </h1>
      <img
        alt="fearow"
        src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/1.png"
      />
      <p>
        Weight: 
      </p>
      <p>
        Price: 
        4
         â‚¬
      </p>
    </div>
  </div>
</div>
`;
